### ISSUE TITLE
Run multiple cargo tasks in parallel

### ISSUE THREAD

**Issue by sourcefrog** [sourcefrog/cargo-mutants#39]
Although cargo and the test framework try to use many cores, they often go through phases of being single-threaded.

The overall mutation tests would go faster if we ran multiple cargo builds in parallel.

Conceivably up to NCPUs builds could be run in parallel. But on some machines that might cause thrashing or too much memory pressure. Possibly we should also look at available memory. There should at least be an option to limit it. As a starting point perhaps NCPUs/2 is reasonable.

This requires making multiple scratch directories. However, we should not make more than the number of discovered mutants.

---

- [x] Factor out a `CargoTask` with a `poll` method
- [x] Launch a thread for each worker
- [x] Add a command-line option `--jobs, -j` and thread it through.
- [x] For now, still default to 1 job?
- [x] Make multiple build directories
- [x] Give the progress model a concept of multiple cargo tasks being underway
- [x] In the `lab` code: 
  - While there's a build dir free and still work to be done, start a new job in it
  - Sleep a fraction of a second
  - Update the progress bar for all running jobs
  - Poll all jobs to see if any have finished
- [x] Update readme docs
- [x] Add at least a smoke test that `-j2` works.
- [x] Test `Mutant::unapply`.
- [x] Clean up self-mutants.

Deferred until later:
- Consider increasing the auto-set timeout when there are multiple jobs?
- #94

**Comment by sourcefrog** [comment]
@dmoka the code in my `process` branch is probably ready to try out if you like, with say `-j4` in some tree.

**Comment by dmoka** [comment]
Sorry, I had a hectic last week and could not find time for this! I just see that it is merged, if you still need to test it, let me know, happy to do that.

**Comment by sourcefrog** [comment]
No problem! It's in the new 1.1.0 release, give it a try and let me know.

**Comment by dmoka** [comment]
Just tested, works like charm, great job! Also read the corresponding docs. 

It will help a lot performance-wise to produce more mutants and running tests against them.

In the doc you say that we can set the `jobs` up to the number of CPU cores. It is not up to the number of threads, so cpu_cores * 2?
